lib/core/src/chain/liquid.rs

```diff
@@ -61,35 +65,49 @@ impl LiquidChainService for HybridLiquidChainService {
-        let tx_bytes = tx.serialize();
-        info!("Broadcasting Liquid tx: {}", tx_bytes.to_hex());
-        let client = reqwest::Client::new();
-        let response = client
-            .post(format!("{LIQUID_ESPLORA_URL}/tx"))
-            .header("Swap-ID", swap_id.unwrap_or_default())
-            .body(tx_bytes.to_hex())
-            .send()
-            .await?;
-        let txid = Txid::from_str(&response.text().await?)?;
-        Ok(txid)
+        match self.network {
+            LiquidNetwork::Mainnet => {
+                let tx_bytes = tx.serialize();
+                info!("Broadcasting Liquid tx: {}", tx_bytes.to_hex());
+                let client = reqwest::Client::new();
+                let response = client
+                    .post(format!("{LIQUID_ESPLORA_URL}/tx"))
+                    .header("Swap-ID", swap_id.unwrap_or_default())
+                    .body(tx_bytes.to_hex())
+                    .send()
+                    .await?;
+                let txid = Txid::from_str(&response.text().await?)?;
+                Ok(txid)
+            }
+            LiquidNetwork::Testnet => Ok(self.electrum_client.broadcast(tx)?),
+        }
     }
 
     async fn get_transactions(&self, txids: &[Txid]) -> Result<Vec<Transaction>> {
         Ok(self.electrum_client.get_transactions(txids)?)
     }
 
     async fn get_script_history(&self, script: &Script) -> Result<Vec<History>> {
-        let script = lwk_wollet::elements::bitcoin::Script::from_bytes(script.as_bytes());
-        let script_hash = sha256::Hash::hash(script.as_bytes())
-            .to_byte_array()
-            .to_hex();
-        let url = format!("{}/scripthash/{}/txs", LIQUID_ESPLORA_URL, script_hash);
-        // TODO must handle paging -> https://github.com/blockstream/esplora/blob/master/API.md#addresses
-        let response = get_with_retry(&url, 3).await?;
-        let json: Vec<EsploraTx> = response.json().await?;
-
-        let history: Vec<History> = json.into_iter().map(Into::into).collect();
-        Ok(history)
+        match self.network {
+            LiquidNetwork::Mainnet => {
+                let script = lwk_wollet::elements::bitcoin::Script::from_bytes(script.as_bytes());
+                let script_hash = sha256::Hash::hash(script.as_bytes())
+                    .to_byte_array()
+                    .to_hex();
+                let url = format!("{}/scripthash/{}/txs", LIQUID_ESPLORA_URL, script_hash);
+                // TODO must handle paging -> https://github.com/blockstream/esplora/blob/master/API.md#addresses
+                let response = get_with_retry(&url, 3).await?;
+                let json: Vec<EsploraTx> = response.json().await?;
+
+                let history: Vec<History> = json.into_iter().map(Into::into).collect();
+                Ok(history)
+            }
+            LiquidNetwork::Testnet => {
+                let mut history_vec = self.electrum_client.get_scripts_history(&[script])?;
```

Comment:
There should be a `electrum_client.script_get_history(script)` to get a single script history

Reply:
We are using ElectrumClient from lwk which wraps the electrum_client.Client which doesn't expose the inner client.
Do you think it worth saving also the Client on the struct only for this call?

Reply:
No, I just thought it would be a bit cleaner